### ISSUE TITLE
In the B&P python example, comparing node value against the current best objective should be done only when the RMP is feasible.

### ISSUE THREAD

**Issue by guifcoelho** [ERGO-Code/HiGHS#1496]
Hello, I was adapting the B&P code and I realised something.

At lines 370-371, if node value is greater than the current best objective then this node is pruned:

```python
if node.value >= best_obj:
     continue
```
However, this will probably raise an error when the master problem is infeasible. `node.value` is only set when the master problem has the status of `HighsModelStatus.kOptimal` (lines 297-301).

It would be best to do it at line 374:

```python
if mp_is_feasible and node.value <= best_obj:
...
```

**Comment by jajhall** [comment]
Thanks. I can probably make this change myself, but could you express it as a PR, please

**Comment by guifcoelho** [comment]
Done!

**Comment by jajhall** [comment]
Fixed by #1497
